lib/ohai/plugins/linux/cpu.rb

```diff
@@ -80,5 +80,6 @@
     cpu cpuinfo
     cpu[:total] = cpu_number
     cpu[:real] = real_cpu.keys.length
+    cpu[:cores] = real_cpu.keys.length * cpu["0"]["cores"].to_i
```

Comment:
Yeah I think its fairly unlikely for a Linux box to have a mix of say dual core and quad core in different sockets.  That may, in fact, make lots of MBs sad if the CPUs aren't reasonably identicalish.  Certainly, we can fix it if it ever becomes a problem, YAGNI.